### PR TITLE
Parallel CI

### PR DESCRIPTION
**Pull request by simi**
This is first step to make CI faster. Currently it shrinks time from 11/12 minutes to 8/9 minutes. General tests could be run in parallel using https://github.com/grosser/parallel_tests (plan to follow if this is merged). My target is ~5 minutes in the end.

@segiddins do you think this is worth it? I'll update checks after if merged.

### CODE REVIEW

File: .github/workflows/test.yml
Comment (segiddins) [inline .github/workflows/test.yml:32]: probably makes sense to move the checkout step into the new action

File: .github/workflows/test.yml
Comment (segiddins) [inline .github/workflows/test.yml:21]: it will be easier to manage the required checks if we add a new job that `needs` this test job (to act as a fan-in for test success)

File: .github/actions/setup-rubygems.org/action.yml
Comment (segiddins) [inline .github/actions/setup-rubygems.org/action.yml:1]: does extracting this into an action actually help make things faster?

Comment (simi) [reply .github/actions/setup-rubygems.org/action.yml:1]: No, it doesn't not. This change is not directly related, but makes it a little easier to maintain workflows. My original plan was to use multiple workflows reusing this shared action, but in the end I found out it is still simple enough to use matrix within one workflow job. For now I can revert this change and keep it locally unless it will be really beneficial.

Comment (segiddins) [reply .github/actions/setup-rubygems.org/action.yml:1]: nah splitting it out is fine, just wanted to understand

File: .github/workflows/test.yml
Comment (simi) [reply .github/workflows/test.yml:32]: IMHO checkout is not part of "setup", but I can move it in there if you feel strongly about this.

File: .github/workflows/test.yml
Comment (simi) [reply .github/workflows/test.yml:21]: I'm not I do fully follow in here. Do you recommend to create new test job with `needs: [rails]` doing nothing in the end (it can just print "tests passing", I'm not sure empty job is supported) and use that one in checks on GH side for required checks setup?

Comment (segiddins) [reply .github/workflows/test.yml:21]: yup, exactly that!

Comment (simi) [reply .github/workflows/test.yml:21]: I'll try.

Comment (simi) [reply .github/workflows/test.yml:28]: @segiddins feel free to check. My plan is to add comment like this to explain.

```
# This is umbrella job for all tests needed to pass to make it simpler
# to maintain GitHub Actions status required checks since job names and
# amount change over the time and it is easier to maintian having just
# this umbrella check set as required.
```

Feel free to propose better wording if needed.

**Review by segiddins — APPROVED** [review]

**Review by martinemde — APPROVED** [review]
Looks good, I think it's totally worth it to aim for 5 minutes. Sometimes waiting for CI to build on a branch is a full blocker for doing something. I like the extracted setup-rubygems.org.

File: .github/workflows/test.yml
Comment (martinemde) [inline .github/workflows/test.yml:17]: Other examples I see just say,
```suggestion
    needs: rails
```
It seems not to be completing though all the other test runs are complete.

Comment (simi) [reply .github/workflows/test.yml:17]: This works as intended. We just need to change repo settings to check for this new check once merged.

Comment (simi) [reply .github/workflows/test.yml:17]: I have merged this and updated the check.

**Pull request merged**
